libraries/node-core-library/src/FileSystem.ts

```diff
@@ -785,6 +791,59 @@ export class FileSystem {
     });
   }
 
+  /**
+   * Writes the contents of multiple Uint8Arrays to a file on disk, overwriting the file if it already exists.
+   * Behind the scenes it uses `fs.writevSync()`.
+   * This API is useful for writing large files efficiently, especially if the input is being concatenated from
+   * multiple sources.
+   * @remarks
+   * Throws an error if the folder doesn't exist, unless ensureFolder=true.
+   * @param filePath - The absolute or relative path of the file.
+   * @param contents - The content that should be written to the file.
+   * @param options - Optional settings that can change the behavior.
+   */
+  public static writeBuffersToFile(
+    filePath: string,
+    contents: Iterable<Uint8Array>,
+    options?: IFileSystemWriteBinaryFileOptions
+  ): void {
+    FileSystem._wrapException(() => {
+      const toCopy: Uint8Array[] = [...contents];
+
+      let fd: number | undefined;
+      try {
+        fd = fsx.openSync(filePath, 'w');
+      } catch (error) {
+        if (!options?.ensureFolderExists || !FileSystem.isNotExistError(error as Error)) {
+          throw error;
+        }
+
+        const folderPath: string = nodeJsPath.dirname(filePath);
+        FileSystem.ensureFolder(folderPath);
+        fd = fsx.openSync(filePath, 'w');
+      }
+
+      let position: number = 0;
+      try {
+        // In practice this loop will have exactly 1 iteration, but the spec allows
```

Comment:
This comment is misleading/confusing -- I think you meant *have exactly 1 iteration per array element*.

Reply:
Comment is correct, in practice the loop will have exactly 1 iteration, since the underlying API processes `toCopy` in its entirety in a single transaction.